src/verify_markup.rs

```diff
@@ -0,0 +1,548 @@
+//! Verification of comment markup and other niceties for HTML generation.
+//!
+//! This is a clone of metamath.exe's `VERIFY MARKUP` command, although the
+//! implementation is somewhat more robust. The intention is to use these
+//! checks for checking the main databases that appear on the Metamath website,
+//! such as [set.mm](https://github.com/metamath/set.mm/). As such, they are
+//! rather more opinionated than the usual checks (for example, labels should
+//! not use underscores), and they also include a number of set.mm-specific
+//! references, because this is what metamath.exe did. (Ideally, most or all of
+//! these hardcoded references can be replaced by `$j` commands in the future.)
+
+use crate::comment_parser::{is_text_escape, CommentItem, CommentParser, Date, Parenthetical};
+use crate::diag::{BibError, Diagnostic};
+use crate::parser::{HeadingComment, HeadingLevel};
+use crate::scopeck::Hyp;
+use crate::segment::{Comparer, SegmentRef};
+use crate::segment_set::SourceInfo;
+use crate::statement::{StatementAddress, NO_STATEMENT};
+use crate::util::{HashMap, HashSet};
+use crate::{Database, Span, StatementRef, StatementType};
+use regex::bytes::Regex;
+use std::ops::Range;
+
+lazy_static::lazy_static! {
+    static ref WINDOWS_RESERVED_NAMES: Regex =
+        Regex::new("(?i-u)^(?:CON|PRN|AUX|NUL|(?:COM|LPT)[1-9])$").unwrap();
+}
```

Comment:
Wow! I suppose that is inherited from Metamath-exe?
Even though this is not a big constraint, I don't think we shall limit ourselves anymore because of these. Anyway I don't think this discussion belongs here. I want to look at the implementation rather than the tests themselves.

Reply:
Yes, this is for compatibility with metamath.exe's verify markup. It is, however, defending against an issue that remains today: I just tested creating `con.html` in Windows 10 and you get the error "The device name is invalid". I think this is a filesystem limitation.

I think a better solution is to not require the HTML files to be `label.html` without modification; this creates lots of artificial limitations like the inability to have `mm*` labels. But this can't be changed as long as metamath-exe remains the way it is now.